docs/pages/versions/unversioned/sdk/sqlite-next.mdx

```diff
@@ -166,23 +164,61 @@ const styles = StyleSheet.create({
 });
 ```
 
+### `useSQLiteContext()` hook with `React.Suspense`
+
+As with the [`useSQLiteContext()` hook](#usesqlitecontext-hook), you can also integrate the `SQLiteProvider` with [`React.Suspense`](https://react.dev/reference/react/Suspense) to show a fallback component until the database is ready.
+To enable the integration, you need to pass the `useSuspense` prop to the `SQLiteProvider` component.
+
+```tsx
+import { SQLiteProvider, useSQLiteContext } from 'expo-sqlite/next';
+import { Suspense } from 'react';
+import { View, Text, StyleSheet } from 'react-native';
+
+export default function App() {
+  return (
+    <View style={styles.container}>
+      <Suspense fallback={<Fallback />}>
+        <SQLiteProvider databaseName="test.db" onInit={migrateDbIfNeeded} useSuspense>
+          <Header />
+          <Content />
+        </SQLiteProvider>
+      </Suspense>
+    </View>
+  );
+}
+```
+
 ### Executing queries within an async transaction
 
 ```js
-const db = await SQLite.openDatabaseAsync('dbName');
+const db = await SQLite.openDatabaseAsync('databaseName');
 
-await db.transactionAsync(async () => {
+await db.withTransactionAsync(async () => {
   const result = await db.getAsync('SELECT COUNT(*) FROM USERS');
   console.log('Count:', result.rows[0]['COUNT(*)']);
 });
 ```
 
-> **Note:** Due to the nature of the async/await, we can't ensure that the transaction will be executed in the same order as the statements are written. A query out of a transaction may be executed inside the transaction. If you care about the order of execution, use the [`transactionExclusiveAsync`](#transactionexclusiveasynctask) function.
+> **Note:** Due to the nature of the async/await, we can't ensure that the transaction will be executed in the same order as the statements are written. A query out of a transaction may be executed inside the transaction. For example, the following test case may fail from out of order async/await statements. If you care about the order of execution, use the [`withTransactionExclusiveAsync`](#withtransactionexclusiveasynctask) function.
+> ```tsx
+> db.withTransactionAsync(async () => {
+>   await db.execAsync('UPDATE test SET name = "aaa"');
+>
+>   //
+>   // We cannot control the order of async/await order, so order of execution is not guaranteed.
+>   // The following UPDATE query out of transaction may be executed here and break the expectation.
+>   //
+>
+>   const result = await db.getAsync<{ name: string }>('SELECT name FROM Users');
+>   expect(result?.name).toBe('aaa');
+> });
+> db.execAsync('UPDATE test SET name = "bbb"');
+> ```
```

Comment:
Suggested some rewording to talk more about transaction isolation:

```suggestion
> **Note:** Due to the nature of async/await, any query that runs while the transaction is active will be included in the transaction. This includes query statements that are outside of the scope function passed to `withTransactionAsync` and may be surprising behavior.

For example, the following test case runs queries inside and outside of a scope function passed to `withTransactionAsync`. However, all of the queries will run within the actual SQL transaction because the second `UPDATE` query runs before the transaction finishes.

> ```tsx
> Promise.all([
>   // 1. A new transaction begins
>   db.withTransactionAsync(async () => {
>     // 2. The value "first" is inserted into the test table and we wait 2
>     //    seconds
>     await db.execAsync('INSERT INTO test (data) VALUES ("first")');
>     await sleep(2000);
>
>     // 4. Two seconds in, we read the latest data from the table
>     const row = await db.getAsync<{ data: string }>('SELECT data FROM test');
>
>     // ❌ The data in the table will be "second" and this expectation will fail.
>     //    Additionally, this expectation will throw an error and roll back the
>     //    transaction, including the `UPDATE` query below since it ran within
>     //    the transaction.
>     expect(row.data).toBe('first');  
>   }),
>   // 3. One second in, the data in the test table is updated to be "second".
>   //    This `UPDATE` query runs in the transaction even though its code is
>   //    outside of it because the transaction happens to be active at the time
>   //    this query runs.
>   sleep(1000).then(async() => db.execAsync('UPDATE test SET data = "second"')),
> ])
> ```

The [`withTransactionExclusiveAsync`](#withtransactionexclusiveasynctask) function addresses this. Only queries that run within the scope function passed to `withTransactionExclusiveAsync` will run within the actual SQL transaction.
```

Reply:
I'd also suggest not putting large code blocks in a callout as callouts are meant to grab attention and convey (most of the times) as a side note/info. I'd suggest making this code block part of the main section and if we really need to hide or provide or provide it as a side note, we can use `Collapsible` in this case.